playbooks/cluster-up.yaml

```diff
@@ -33,76 +40,98 @@
+# OCP deployment
 - hosts: ec2hosts
   name: configuration
-  remote_user:  "{{ ssh_login }}"
+  remote_user: "{{ ssh_login }}"
   gather_facts: True
   become: yes
-  vars:
-    known_hosts_path: /root/.ssh/known_hosts
-    key_path: /root/.ssh/ocp_key
   environment:
-    OCP_CLUSTER_ID: "{{ tag_name }}"
+    ANSIBLE_HOST_KEY_CHECKING: false
     AWS_ACCESS_KEY_ID: "{{ lookup('env', 'AWS_ACCESS_KEY_ID') }}" 
     AWS_ACCESS_SECRET_KEY: "{{ lookup('env', 'AWS_SECRET_ACCESS_KEY') }}"
-    ANSIBLE_HOST_KEY_CHECKING: false
  
   tasks:
   - name: check docker service
     service: name=docker state=started
 
-  - name: Set aws access key
+  - name: send ocp inventory template to host 
+    copy:
+      src: "{{ src_ocp_inventory_file }}" 
+      dest: "{{ dest_ocp_inventory_file }}"
+      owner: root
+      group: root
+      mode: 0644
+
+  - name: configure inventory host
+    replace:
+      path: "{{ dest_ocp_inventory_file }}"
+      regexp: 'PRIVATE_DNS_NAME'
+      replace: "{{ hostvars[inventory_hostname].private_dns_name }}"
+  
+  - name: export aws access key
     lineinfile:
       path: /root/.bashrc
       regexp: 'export AWS_ACCESS_KEY_ID=.*'
       line: "export AWS_ACCESS_KEY_ID={{ lookup( 'env', 'AWS_ACCESS_KEY_ID') }}"
 
-  - name: set aws secret key
+  - name: export aws secret key
     lineinfile:
       path: /root/.bashrc
       regexp: 'export AWS_SECRET_ACCESS_KEY=.*'
       line: "export AWS_SECRET_ACCESS_KEY={{ lookup( 'env', 'AWS_SECRET_ACCESS_KEY') }}"
 
-  - name: configure ocp
-    blockinfile:
-      path: /root/.bashrc
-      insertafter: EOF
-      block: |
-        export OCP_CLUSTER_ID={{ tag_name }}
-        export ANSIBLE_HOST_KEY_CHECKING=false
+  - name: checkout version branch 3.11
+    shell: cd /root/openshift-ansible && git checkout release-3.11
 
-  - name: copy ocp anible inventory template
-    copy:
-      src: "{{ src_ocp_inventory_file }}" 
-      dest: "{{ dest_ocp_inventory_file }}"
-      owner: root
-      group: root
-      mode: 0644
+  - name: deploy ocp 3.11 (30-45min runtime)
+    shell: cd /root/openshift-ansible && ansible-playbook -i /root/aws-hosts /root/openshift-ansible/playbooks/deploy_cluster.yml
+    register: ocp_cmd
+    ignore_errors: true 
 
-  - name: configure inventory
-    replace:
-      path: "{{ dest_ocp_inventory_file }}"
-      regexp: '(PRIVATE_DNS_NAME)'
-      replace: "{{ hostvars[inventory_hostname].private_dns_name }}"
+  - name: oc deploy report
+    debug: "msg={{ ocp_cmd.stdout_lines }}"
 
-  - name: deploy ocp 3.11 (10~20 minutes)
-    shell: cd /root/openshift-ansible && git checkout release-3.11 && ansible-playbook -i /root/aws-hosts /root/openshift-ansible/playbooks/deploy_cluster.yml
-    register: ocp_cmd
-    no_log: true
-    ignore_errors: true
-    
-  - name: print error
-    debug: "msg={{ ocp_cmd }}"
-    failed_when: ocp_cmd.rc > 0
-   
+  - fail: "msg={{ ocp_cmd.stderr }}"
+    when: ocp_cmd.rc > 0
+
+  - name: configure ocp AiO cluster
```

Comment:
so does this name imply that _cluster_up_ only supports AiO clusters and is not generic for n-node clusters?

Reply:
Yes.  This PB can only deploy single node, AiO clusters. It's just that it can deploy a lot of them at once.